lib/omnibus/fetchers/git_fetcher.rb

```diff
@@ -259,7 +267,10 @@ def self.revision_from_remote_reference(ref, source)
       # allows us to return the SHA of the tagged commit for annotated
       # tags. We take care to only return exact matches in
       # process_remote_list.
-      remote_list = shellout!("git ls-remote \"#{source[:git]}\" \"#{ref}*\"").stdout
+      remote_list = retry_block("git ls-remote", [CommandTimeout, CommandFailed]) do
```

Comment:
Similar to my comment above...knowing the source would that `git ls-remote` is calling would be helpful for debug situations.